btrack/_localization.py

```diff
@@ -211,12 +216,20 @@ def segmentation_to_objects(
             # concatenate the centroids
             centroids = _concat_centroids(centroids, _centroids)
 
-    elif inspect.isgeneratorfunction(segmentation) or isinstance(
-        segmentation, Generator
-    ):
+    else:
 
-        for frame, seg in enumerate(segmentation):
-            intens = next(intensity_image) if USE_INTENSITY else None
+        if segmentation.ndim not in (3, 4):
+            raise ValueError("Segmentation array must have 3 or 4 dims.")
```

Comment:
Might be overkill could `3` and `4` not be magic numbers, i.e. `3D, 4D = (3, 4)`

Reply:
I would do this, but I don't think vars can start with a number, so it'd have to be `three_d` or similar?

Reply:
Ah yeah good point

Reply:
I added a frozen dataclass `Dimensionality` so that one can specify this as `Dimensionality.TWO` and so forth. Updated this in another spot in the code too to reuse it.  How's that?

Reply:
Looks good!